include/ddc/kernels/fft.hpp

```diff
@@ -176,6 +176,18 @@ void rescale(
             });
 }
 
+template <class DDim>
+Real forward_full_norm_coef(DiscreteDomain<DDim> const& ddom) noexcept
```

Comment:
```suggestion
template <class DDim>
auto forward_full_norm_coef(DiscreteDomain<DDim> const& ddom) noexcept -> DiscreteDomain<DDim>::discrete_element_type::value_type
```

I may just prefer `auto`. Maybe with a guidance?

Reply:
In DDC we try to avoid `auto` unless there is a `if constexpr` logic or an unreadable type.

That being said, i don't see how it helps to understand the return type of the function. I find it even less clear there will be an implicit conversion.

Reply:
Sure. Then, that is fine
My point is to avoid a global type and allow users to choose a precision

Reply:
They should already be able to pick the types they want, even mixing the types. I don't really know how to address the latter case. There are different solutions and i don't how to decide which one is the best. But this discussion goes out of scope of the current PR.

Reply:
What do you think of this new version ?